Jenkinsfile.public

```diff
@@ -90,6 +90,23 @@ pipeline {
                         sh 'make test-sim'
                     }
                 }
+
+                stage('Backwards Compatibility') {
+                    when {
```

Comment:
You can use:
```
when {
    branch 'master'
}
```

Reply:
i tried that here https://github.com/storj/storj/pull/2774/commits/4cab79887a3da97ed08f2199b432d201c16cec36, but it didn't run the tests. https://build.ops.storj.io/blue/organizations/jenkins/storj/detail/PR-2774/3/pipeline

Reply:
Hmm, weird. I wonder if it's handling PR-s with different branch names.

Reply:
i don't have a preference to run it for master or all PRs.

Reply:
Let's run it for all, until it starts causing problems.

Reply:
One thing we could test is to run it after `make test-sim` in Integration. That would mean it'll use less concurrency and might be faster.